### ISSUE TITLE
[Bug] Use of zip leads to silent incomplete evaluation of IndependentModelList

### ISSUE THREAD

**Issue by saitcakmak** [cornellius-gp/gpytorch#2211]
# 🐛 Bug

Due to the use of `zip`, when fewer than # of models inputs are passed to `IndependentModelList.__call__/forward` (or even `fantasize`), it will only evaluate the first # of inputs models and ignore the rest. 

## To reproduce

This extracted from the unit tests for `IndependentModelList`, which also falls into this bug.

** Code snippet to reproduce **
```python
import gpytorch
from gpytorch.models import IndependentModelList


class ExactGPModel(ExactGP):
    def __init__(self, train_x, train_y, likelihood):
        super().__init__(train_x, train_y, likelihood)
        self.mean_module = gpytorch.means.ConstantMean()
        self.covar_module = gpytorch.kernels.ScaleKernel(gpytorch.kernels.RBFKernel())

    def forward(self, x):
        mean_x = self.mean_module(x)
        covar_x = self.covar_module(x)
        return gpytorch.distributions.MultivariateNormal(mean_x, covar_x)


N_PTS = 5


def create_test_data():
    return torch.randn(N_PTS, 1)


def create_likelihood_and_labels():
    likelihood = gpytorch.likelihoods.GaussianLikelihood()
    labels = torch.randn(N_PTS) + 2
    return likelihood, labels


def create_model():
    data = create_test_data()
    likelihood, labels = create_likelihood_and_labels()
    return ExactGPModel(data, labels, likelihood)

models = [create_model() for _ in range(2)]
model = IndependentModelList(*models)
model.eval()
model(torch.rand(3))
# This outputs [MultivariateNormal(loc: torch.Size([3]))], which is the evaluation of the first model only.

model(torch.rand(3), torch.rand(3))
# This outputs [MultivariateNormal(loc: torch.Size([3])), MultivariateNormal(loc: torch.Size([3]))], which is both models.

```

## Expected Behavior

Either error out when the # of inputs does not match # of models, or repeat the input if only a single tensor input is given. 
I'd be happy to patch this if we agree on a solution.

## System information

**Please complete the following information:**
-  GPyTorch: Latest
-  PyTorch: Latest
-  OS: CentOS8

## Additional context
Originally surfaced in https://github.com/pytorch/botorch/issues/1467

**Comment by Balandat** [comment]
I think repeating the input if only one is an interesting idea - it's convenient as an API, but also potentially dangerous in that it could mask user errors. By requiring the same number of outputs we'd make sure people understand what this is doing.
